tests/integrationv2/test_renegotiate.py

```diff
@@ -0,0 +1,259 @@
+import copy
+import os
+import pytest
+import time
+import random
+from collections import namedtuple
+
+from configuration import available_ports, ALL_TEST_CIPHERS, ALL_TEST_CURVES, MINIMAL_TEST_CERTS, PROTOCOLS
+from common import ProviderOptions, Protocols, Curves
+from fixtures import managed_process
+from providers import Provider, S2N, OpenSSL, GnuTLS
+from utils import invalid_test_parameters, get_parameter_name
+
+
+# TLS1.3 does not support renegotiation
+TEST_PROTCOLS = [x for x in PROTOCOLS if x.value < Protocols.TLS13.value]
+
+
+# Command line options to enable renegotiation
+S2N_RENEG_OPTION = "--renegotiation"
+S2N_RENEG_ACCEPT = "accept"
+S2N_RENEG_REJECT = "reject"
+
+
+# Output indicating renegotiation state
+S2N_RENEG_START_MARKER = "RENEGOTIATE"
+S2N_RENEG_SUCCESS_MARKER = "s2n is ready, again"
+OPENSSL_RENEG_REQ_MARKER = "SSL_accept:SSLv3/TLS write hello request"
+OPENSSL_RENEG_WARN_MARKER = "SSL3 alert read:warning:no renegotiation"
+
+
+# Methods to check renegotiation state
+def renegotiate_was_requested(openssl_results):
+    return to_bytes(OPENSSL_RENEG_REQ_MARKER) in openssl_results.stderr
+def renegotiate_was_rejected(openssl_results):
+    return to_bytes(OPENSSL_RENEG_WARN_MARKER) in openssl_results.stderr
+def renegotiate_was_started(s2n_results):
+    return to_bytes(S2N_RENEG_START_MARKER) in s2n_results.stdout
+def renegotiate_was_successful(s2n_results):
+    return renegotiate_was_started(s2n_results) and \
+           to_bytes(S2N_RENEG_SUCCESS_MARKER) in s2n_results.stdout
+
+
+# Basic conversion methods
+def to_bytes(val):
+    return str(val).encode('utf-8')
+def to_marker(val):
+    return bytes(val).decode('utf-8')
+
+
+"""
+The integrationv2 framework controls process IO with "markers".
```

Comment:
I'm not sure if this paragraph gives me any useful information to understand what Msg is used for.